org.opent2t.sample.multisensor.superpopular/com.wink.sensorpod/js/thingTranslator.js

```diff
@@ -0,0 +1,451 @@
+'use strict';
+var OpenT2TError = require('opent2t').OpenT2TError;
+
+// This code uses ES2015 syntax that requires at least Node.js v4.
+// For Node.js ES2015 support details, reference http://node.green/
+
+/**
+ * Validates an argument matches the expected type.
+ */
+function validateArgumentType(arg, argName, expectedType) {
+    if (typeof arg === 'undefined') {
+        throw new OpenT2TError(400, 'Missing argument: ' + argName + '. ' +
+            'Expected type: ' + expectedType + '.');
+    } else if (typeof arg !== expectedType) {
+        throw new OpenT2TError(400, 'Invalid argument: ' + argName + '. ' +
+            'Expected type: ' + expectedType + ', got: ' + (typeof arg));
+    }
+}
+
+/**
+ * Finds a resource for an entity in a schema
+ */
+function findResource(schema, di, resourceId) {
+    // Find the entity by the unique di 
+    var entity = schema.entities.find((d) => {
+        return d.di === di;
+    });
+
+    if (!entity) {
+        throw new OpenT2TError(404, 'Entity - ' + di + ' not found.');
+    }
+
+    var resource = entity.resources.find((r) => {
+        return r.id === resourceId;
+    });
+
+    if (!resource) {
+        throw new OpenT2TError(404, 'Resource with resourceId \"' + resourceId + '\" not found.');
+    }
+    
+    return resource;
+}
+
+/**
+ * Wink does not always populate every desired_state property, but last_reading doesn't necessarily
+ * update as soon as we send our PUT request. Instead of relying just on one state or the other,
+ * we use this StateReader class to read from desired_state if it is there, and fall back to last_reading
+ * if it is not.
+ */
+class StateReader {
+    constructor(desired_state, last_reading) {
+        this.desired_state = desired_state;
+        this.last_reading = last_reading;
+    }
+
+    get(state) {
+        if (this.desired_state[state] !== undefined) {
+            return this.desired_state[state];
+        }
+        else {
+            return this.last_reading[state];
+        }
+    }
+
+    containsKey(state) {
+        return (state in this.desired_state ||
+                state in this.last_reading); 
+    }
+}
+
+function convertDeviceDateToTranslatorDate(unixTimestamp) {
+    // Date takes a number of milliseconds, so convert seconds to milliseconds
+    var datetime = new Date(unixTimestamp * 1000);
+
+    return datetime.toISOString();
+}
+
+function convertDeviceBatteryToTranslatorBattery(batteryValue) {
+    return Math.round(batteryValue * 100);
+}
+
+/**
+ * Returns a default value if the specified property is null, undefined, or an empty string
+ */
+function defaultValueIfEmpty(property, defaultValue) {
+    if (property === undefined || property === null || property === "") {
```

Comment:
The same function is used in the other Wink translators. The one concern I have with using if (property) is that it would make defaultValueIfEmpty not work for Boolean values (since that would evaluate to false if the property read from the schema is actually false).

Unless you think this is urgent, I suggest leaving it as is and changing all of the Wink translators at once in a separate PR if it's an issue.